harness/tests/determined/common/experimental/test_experiment.py

```diff
@@ -229,3 +229,61 @@ def test_download_code_writes_output_to_file(
     with open(output_file, "rb") as f:
         file_content = f.read()
         assert file_content == b"b64TgzResponse"
+
+
+@mock.patch("determined.common.api.bindings.get_GetExperimentCheckpoints")
+def test_list_checkpoints_calls_bindings_with_sort_by_metric(
+    mock_bindings: mock.MagicMock,
+    make_expref: Callable[[int], experiment.Experiment],
+) -> None:
+    expref = make_expref(1)
+    ckpt_resp = api_responses.sample_get_experiment_checkpoints()
+    mock_bindings.side_effect = api_responses.iter_pages(
+        pageable_resp=ckpt_resp,
+        pageable_attribute="checkpoints",
+    )
+
+    sort_by_metric = "val_metric"
+    expref.list_checkpoints(sort_by=sort_by_metric, order_by=checkpoint.CheckpointOrderBy.ASC)
+
+    _, call_kwargs = mock_bindings.call_args_list[0]
+
+    assert call_kwargs["sortByMetric"] == sort_by_metric
+
+
+@mock.patch("determined.common.api.bindings.get_GetExperimentCheckpoints")
+def test_list_checkpoints_calls_bindings_with_sort_by_attr(
+    mock_bindings: mock.MagicMock,
+    make_expref: Callable[[int], experiment.Experiment],
+) -> None:
+    expref = make_expref(1)
+    ckpt_resp = api_responses.sample_get_experiment_checkpoints()
+    mock_bindings.side_effect = api_responses.iter_pages(
```

Comment:
Is this actually used in the test?

Reply:
yeah, it needs to be mocked otherwise `get_with_offset` will throw up.

Reply:
This didn't make any sense to me, so I some debugging. It's not `get_with_offset` that's throwing up, it's `mock`. [mock expects](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.side_effect) a `side_effect` to be:
> a function to be called when the mock is called, an iterable or an exception (class or instance) to be raised.

You can fix this by
```python
mock_bindings.side_eddect = [ckpt_resp]
```
or
```python
mock_bindings.return_value = ckpt_resp
```

Reply:
Oh! Retroactively realizing that my original comment was ambiguous because it's attached to the entire line and not just the `iter_pages` part.

Reply:
nice, thanks